client/devdocs/design/illustrations.jsx

```diff
@@ -0,0 +1,227 @@
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import Main from 'components/main';
+
+/**
+ * Illustrations
+ */
+import adwordsGoogle from 'assets/images/illustrations/adwords-google.svg';
```

Comment:
> I wanted to add something that allows you to click an image to copy the line you'd import in a new project, but I haven't figured out how to both import the assets via webpack and keep the private URL intact

Something that may be worth exploring is to replace all these imports with dynamic imports by iterating a object containing the paths to import:

```js
const illustrations = [
  {
    name: 'General',
    path: null,
    images: [
      'app-banner-editor',
      'app-banner-notifications',
      // ...
    ],
  },
  {
    name: 'Customer Home',
    path: 'custome-home',
    images: [
      'illustration--business',
      'illustration--checklist-complete',
      // ...
    ],
  },
]

const Illustrations = () => (
  <Main className="devdocs design__illustrations devdocs__illustrations">
    <DocumentHead title="Illustrations" />
	  <div className="design__illustrations-content devdocs__doc-content">
        <h1>Illustrations</h1>
        <p></p>
        { illustrations.map( ( category ) => (
          <h2>{ category.name }</h2>
          <div className="design__illustrations-illustrations">
            { category.images.map( ( imagePath ) => {
              const path = `assets/images/illustrations/${ category.path ? category.path + '/' : '' }${ imagePath }.svg`;
              let image = await import( path );
              return (
                <div className="design__illustrations-illustration">
                  <img src={ image } alt="" onClick={ () => copyToClipboard( `import ${ camelCaseToSlug( image ) } from '${ path }';` ) } />
                </div>
              );
            } ) }
          </div>
        ) ) }
      </div>
  </DocumentHead>
  </Main>
);
```

Reply:
I started down this path at one point, but worried it would have the same effect as passing from a prop. Although I suppose if the array is static and we know the objects exist, it wouldn't?

Reply:
I started down this path, but worried we'd end up with the same problem we had with dynamic imports and webpack; but I suppose it's not an issue if we're using a static array rather than a dynamic prop?

Reply:
Even if webpack cannot determine what are the images we import and all of them are bundled, it wouldn't matter given that we want to render all illustrations here, right? I think the bundle size for this new section in the dev docs shouldn't be affected.

Reply:
In devdocs, I think doing the `images/illustrations/*` import is OK. We want to bundle all of them anyway.

It's only in a general purpose `<Illustration />` component, intended to be used in app UI, where such an import would deoptimize the bundle and ship a lot of wasted code.

---

The `await import( path )` in a `render` function will not work -- the `render` function is not async. You'll need to do something like:
```js
function Illustrations( { category } ) {
  const [ srcs, setSrcs ] = useState();
  useEffect( () => {
    const srcsLoad = category.images.map( async path => {
      return ( await import( /* webpackMode: "eager" */, path ) ).default;
    } );
    Promise.all( srcsLoad ).then( loadedSrcs => setSrcs( loadedSrcs ) );
  }, [ category ] );

  if ( ! srcs ) {
    return null; // or some placeholder
  }
  
  return (
    <div>
      { srcs.map( src => <img src={ src } /> ) }
    </div>
  );
}
```
I.e., use local state and an effect to do the async load.

The `webpackMode: "eager"` comment will prevent creating a new chunk for each image module. They will all be merged into the parent chunk, i.e., `devdocs`. The dynamic `import( path )` will not trigger a network request (the `devdocs` chunk is already loaded), and will just return a promise that resolves immediately.

Creating a new chunk for each image would unnecessarily inflate the number of webpack chunks and inflate the size of the runtime manifest that every Calypso request (even a non-devdocs one) must load.